### PR TITLE
dateutils: new package

### PR DESCRIPTION
**Pull request by TheQueasle**
> Dateutils are a bunch of tools that revolve around fiddling with dates and times in the command line with a strong focus on use cases that arise when dealing with large amounts of financial data. 

- http://www.fresse.org/dateutils/
- https://github.com/hroptatyr/dateutils

### CODE REVIEW

File: var/spack/repos/builtin/packages/dateutils/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    version('0.4.6', sha256='26a071317ae5710f226a3e6ba9a54d3764cd9efe3965aecc18e75372088757cd')
+    version('0.4.5', sha256='16d6a0fe7b7d49ddbb303f33538dd7304a0d4af5a0369bcbf275db6a5060cbde')
+
+    build_directory = 'spack-build'
+
+    depends_on('texinfo')
+    depends_on('gperf')
+    depends_on('flex')
+    depends_on('bison')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dateutils/package.py]: These last two are often `build` dependencies only. Can you use `ldd -r` (on Linux) or `otool -L` (on macOS) to see if they are actually link dependencies too?

Comment (TheQueasle) [reply var/spack/repos/builtin/packages/dateutils/package.py]: Sure thing!

An `ldd -r` of one of the compiled binaries:
```
$ ldd -r $(which dateseq)
        linux-vdso.so.1 =>  (0x00002aaaaaacd000)
        libdl.so.2 => /usr/lib64/libdl.so.2 (0x00002aaaaaccf000)
        libc.so.6 => /usr/lib64/libc.so.6 (0x00002aaaaaed3000)
        /lib64/ld-linux-x86-64.so.2 (0x00002aaaaaaab000)
```

Looking at that...it does not appear to be pulling any of those from a spack install, as I would expect to see the install path of spack listed.

I pulled the requirements from https://github.com/hroptatyr/dateutils/blob/master/INSTALL ... though it looks like I didn't read far enough down the document.  According to it I would only need a compiler if building dateutils from the release tarball.

I'll do some more work on this and update my branch.

Thanks!

Comment (TheQueasle) [reply var/spack/repos/builtin/packages/dateutils/package.py]: Though now that I have done some more poking around, I see that those linked libraries being system libraries may not matter? https://github.com/spack/spack/issues/7445#issuecomment-373483171

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/dateutils/package.py]: Yep, those are all standard system libraries that Spack doesn't manage.

Comment (TheQueasle) [reply var/spack/repos/builtin/packages/dateutils/package.py]: Ok - should be resolved now with my latest commit.

**Pull request merged**